test/NuGet.Core.Tests/NuGet.Packaging.Test/DependenciesGroupsForEachTFMRuleTests.cs

```diff
@@ -0,0 +1,446 @@
+using Xunit;
+using NuGet.RuntimeModel;
+using NuGet.Client;
+using NuGet.ContentModel;
+using NuGet.Packaging.Core;
+
+namespace NuGet.Packaging.Test
+{
+    public class DependenciesGroupsForEachTFMRuleTests
+    {
+        [Fact]
+        public void GenerateWarnings_PackageWithDependenciesForEachTFMInLib_ShouldNotWarn()
+        {
+            //Arrange
+            var frameworks = new NuGetFramework[]
+            {
+                FrameworkConstants.CommonFrameworks.Net2,
+                FrameworkConstants.CommonFrameworks.Net35,
+                FrameworkConstants.CommonFrameworks.Net4,
+                FrameworkConstants.CommonFrameworks.Net45,
+                FrameworkConstants.CommonFrameworks.NetStandard10,
+                FrameworkConstants.CommonFrameworks.NetStandard13,
+                FrameworkConstants.CommonFrameworks.NetStandard20,
+            };
+
+            var fileStrings = new string[]
+            {
+                "lib/net20/test.dll",
+                "lib/net35/test.dll",
+                "lib/net40/test.dll",
+                "lib/net45/test.dll",
+                "lib/netstandard1.0/test.dll",
+                "lib/netstandard1.3/test.dll",
+                "lib/netstandard2.0/test.dll",
+            };
+
+            // Act
+            var rule = new DependenciesGroupsForEachTFMRule();
+            var (compat, file, nuspec) = rule.Categorize(fileStrings, frameworks);
+            var issues = rule.GenerateWarnings(compat, file, nuspec).ToList();
+
+            // Assert
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5128));
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5130));
+        }
+
+        [Fact]
+        public void GenerateWarnings_PackageWithDependenciesInRefHasExactMatch_ShouldNotWarn()
+        {
+            var frameworks = new NuGetFramework[]
+            {
+                FrameworkConstants.CommonFrameworks.Net2,
+                FrameworkConstants.CommonFrameworks.Net35,
+                FrameworkConstants.CommonFrameworks.Net4,
+                FrameworkConstants.CommonFrameworks.Net45,
+                FrameworkConstants.CommonFrameworks.NetStandard10,
+                FrameworkConstants.CommonFrameworks.NetStandard13,
+                FrameworkConstants.CommonFrameworks.NetStandard20,
+            };
+
+            //Arrange
+            var fileStrings = new string[]
+            {
+                "ref/net20/test.dll",
+                "ref/net35/test.dll",
+                "ref/net40/test.dll",
+                "ref/net45/test.dll",
+                "ref/netstandard1.0/test.dll",
+                "ref/netstandard1.3/test.dll",
+                "ref/netstandard2.0/test.dll",
+            };
+
+            // Act
+            var rule = new DependenciesGroupsForEachTFMRule();
+            var (compat, file, nuspec) = rule.Categorize(fileStrings, frameworks);
+            var issues = rule.GenerateWarnings(compat, file, nuspec).ToList();
```

Comment:
Don't worry about making changes if you don't have time, but an FYI for the future, it's more clear if you created the `compat`, `file` and `nuspec` collections in the test and don't call `Categorize`. That way it's more obvious to the reader what the inputs are to `GenerateWarnings`.